python/lsst/ip/diffim/modelPsfMatch.py

```diff
@@ -327,11 +327,11 @@ def run(self, exposure, referencePsfModel, kernelSum=1.0):
         result = self._buildCellSet(exposure, referencePsfModel)
         kernelCellSet = result.kernelCellSet
         referencePsfModel = result.referencePsfModel
-        # TODO: This should be evaluated at (or close to) the center of the
-        # exposure's bounding box in DM-32756.
-        sciAvgPos = exposure.getPsf().getAveragePosition()
+        fwhmScience = diutils.evaluateMeanPsfFwhm(exposure,
+                                                  fwhmExposureBuffer=self.kConfig.fwhmExposureBuffer,
+                                                  fwhmExposureGrid=self.kConfig.fwhmExposureGrid
+                                                  )
         modelAvgPos = referencePsfModel.getAveragePosition()
-        fwhmScience = exposure.getPsf().computeShape(sciAvgPos).getDeterminantRadius()*sigma2fwhm
         fwhmModel = referencePsfModel.computeShape(modelAvgPos).getDeterminantRadius()*sigma2fwhm
```

Comment:
Isn't the problem that the average position may be a position that doesn't evaluate to a finite psf?

Reply:
I believe the problem is for coadds, which might have gaps, rather than for science images. This code is also not used in production.

Reply:
This code isn't used in production?  I thought that this PR originally was inspired by crashes when running by HSC people, but maybe that's not AP production, but people are using it.  
As for the other issue, science images may also have gaps (one of HSC detectors has a bad amp in the middle for example).

Reply:
Sorry, I meant that `modelPsfMatch` (and `imagePsfMatch`) is not used in production. The rest of the code being changed on this PR is, and this PR is correctly accounting for potential gaps in science images in that code.

Reply:
Given that `evaluateMeanPsfFwhm` basically tries `getPsfFwhm` first anyway, isn't it safer to use that instead of using `getPsfFwhm` explicitly?

Reply:
I was requesting elsewhere in this PR to remove `getPsfFwhm` from `evaluateMeanPsfFwhm`, since the calculations are very different and should not be considered interchangeable in general.